### ISSUE TITLE
Use font-awesome-type for Font Awesome icons

### ISSUE THREAD

**Issue by spl** [reflex-frp/reflex-dom-semui#5]
I just created https://github.com/spl/font-awesome-type for Font Awesome icons. Since I've been using Bootstrap, I always found it quite annoying to use strings for the glyphs. So I think it would be better to use some something constructive from the start.

Do you want to use `font-awesome-types` for the Font Awesome icon enumeration? If so, I can work on that. Also, I'd appreciate a review of it before I publish it to Hackage. (It's a pretty straightforward package, but another set of eyes would be nice.)

**Comment by mightybyte** [comment]
Wow, you read our minds!  @imalsogreg and I were just talking about doing this.  We decided we could always add it on later as a layer on top of the String layer.

**Comment by spl** [comment]
Funny! After I pushed `font-awesome-type`, I noticed you did a bit on icons. I figured the Font Awesome stuff was relatively general and independent of `reflex-dom-semui`, so I whipped together a quick package.

**Comment by mightybyte** [comment]
Couldn't have worked out more perfectly.  You should be able to add your font awesome stuff on top of what I did.

**Comment by mightybyte** [comment]
We were also thinking it would be great to get the actual icons to show up in the haddocks.  Something to consider...

**Comment by spl** [comment]
> We were also thinking it would be great to get the actual icons to show up in the haddocks.

I agree. Not sure how to do that currently. These are perhaps the main issues:
1. loading Font Awesome in the Haddock output
2. using HTML in the comments

**Comment by ekmett** [comment]
Depending on your focus you might also consider the [material design icons](https://design.google.com/icons/) from google and possibly even the [Kenney game icons](http://opengameart.org/content/game-icons).

Both are set up as fonts in the same manner as font-awesome, and admit a similar API.

Material design has the benefit of having more consistent widths than font-awesome for the things it covers, but has a largely disjoint range of covered topics.

The following repository supports c headers for all 3:

https://github.com/juliettef/IconFontCppHeaders

**Comment by spl** [comment]
@ekmett: Good suggestion.

It would certainly be useful to allow for different icon sets, so it's important to have widgets that are flexible.

I started with Font Awesome here since Semantic UI [includes Font Awesome](http://semantic-ui.com/elements/icon.html). Considering that fact, I think it makes sense to build support for Font Awesome into `reflex-dom-semui`.

**Comment by mightybyte** [comment]
Thanks for the pointer to those other icons @ekmett.  I think right now I'm going to keep the focus of this package on stuff that is included in Semantic UI.  However it might make sense make a dedicated icon package, `reflex-icons` or something similar that includes all three of those.

**Comment by spl** [comment]
Et [voilà](https://hackage.haskell.org/package/font-awesome-type-0.1/docs/Web-FontAwesomeType.html):

<img width="454" alt="fa-icons-in-docs" src="https://cloud.githubusercontent.com/assets/54900/19340301/7ec6f5be-9128-11e6-991f-b52ce6238b97.png">

**Comment by hSloan** [comment]
Thank you for this package! I was wondering, will you be updating this package with Font Awesome(v 4.7)?

**Comment by mightybyte** [comment]
@hSloan You may want to mention that here instead: https://github.com/spl/font-awesome-type
